### ISSUE TITLE
BME680 integration

### ISSUE THREAD

**Issue by diegoweb100** [arachnetech/homebridge-mqttthing#276]
Hi! Is there an option to make BME680 working with Air Quality Sensor or gas sensor?
It would be good if we can set also the gas type...

I've found different thing online like this:
https://github.com/trho/homebridge-bme680
https://github.com/pawelosin/homebridge-bme680

**Comment by rbswift** [comment]
should be fine to use with Tasmota and something like...

```
                "getAirQuality": {
                    "topic": "tele/tasmota/SENSOR",
                    "apply": "return JSON.parse(message).BME680.Gas;"
                }
```

adjust TelePeriod in Tasmota according to preference

**Comment by rbswift** [comment]
actually, you need to do a bit more work because you also need to classify the values into excellent, good, fair, inferior, and poor in the "apply" javascript function to correspond to whatever you set in airQualityValues

"airQualityValues": [ "unknown-value", "excellent-value", "good-value", "fair-value", "inferior-value", "poor-value" ],

**Comment by diegoweb100** [comment]
Yes this is the problem...to calculate the corresponding ppm value we need to include the BME680 driver by Bosch else we have a resistor value 

I've already implemented something like what you said but obviously without the driver included into homebridge-mqttthing will not work properly..


            "topics": {
                "getAirQuality": {
                    "topic": "tele/GAS/SENSOR",
                    "apply": "gas = JSON.parse(message).BME680.Gas; return (gas > 2000 ? 'POOR' : gas > 1600 ? 'INFERIOR' : gas > 1100 ? 'FAIR' : gas > 700 ? 'GOOD' : gas <= 700 ? 'EXCELLENT' : 'UNKNOWN');"
                },
                "getAirQualityPPM": {
                    "topic": "tele/GAS/SENSOR",
                    "apply": "return JSON.parse(message).BME680.Gas;"
                },
                "getCurrentTemperature": {
                    "topic": "tele/GAS/SENSOR",
                    "apply": "return JSON.parse(message).BME680.Temperature;"
                },
                "getCurrentRelativeHumidity": {
                    "topic": "tele/GAS/SENSOR",
                    "apply": "return JSON.parse(message).BME680.Humidity;"
                },
                "getCarbonDioxideLevel": {
                    "topic": "tele/GAS/SENSOR",
                    "apply": "return JSON.parse(message).BME680.Gas;"
                }
            },
            "airQualityValues": [
                "UNKNOWN",
                "EXCELLENT",
                "GOOD",
                "FAIR",
                "INFERIOR",
                "POOR"
            ],

**Comment by diegoweb100** [comment]
up!

**Comment by arachnetech** [comment]
I'm not sure that I understand: are you saying it's not possible to translate the numeric values for air quality into Homekit's states using a fixed mapping table? Do those BME680 libraries that you reference do a more complex calculation combining the other sensor values with 'gas' in order to compute air quality?

**Comment by diegoweb100** [comment]
Yes exactly, you get the point! Bme680 is a little bit complicated sensor...Also you need to chose with kind of “gas”.

This is what I get from tasmota:
![2DB42113-6AE1-482B-8D25-5E894AE40A7E](https://user-images.githubusercontent.com/42244185/87250843-fd62af00-c467-11ea-998d-a64d4258c316.jpeg)

**Comment by diegoweb100** [comment]
up!

**Comment by arachnetech** [comment]
Sorry, I did see this! I just haven't have time to look. 😀

I think it would be best implemented as a codec, to avoid loading the BME library unnecessarily.

**Comment by diegoweb100** [comment]
ok  how I can do that??
